test/parallel/test-event-emitter-check-bind-functions.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const events = require('events');
+
+{
+  const E = events.EventEmitter.prototype;
+
+  assert.strictEqual(E.on, E.addListener);
+  assert.strictEqual(E.off, E.removeListener);
+
+  E.on = function() {};
+  assert.strictEqual(E.on, E.addListener);
```

Comment:
Why would this ever be the desired behavior? It breaks the principle of least surprise.

Reply:
`.on` is just an alias to `.addListener`. This does prevent some issues where someone wants to change the functionality but forgets to update all functions involved. I guess in this specific case it's fine therefore?